enhancements/cluster-runtime-constraints.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Cluster Runtime Constraint
+authors:
+  - "@dinhxuanvu"
+reviewers:
+  - "@kevinrizza"
+approvers:
+  - TBD
+creation-date: 2021-08-26
+last-updated: 2020-09-06
+status: provisional
+---
+
+# cluster-runtime-constraint
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This proposal presents the specification and usage to enable runtime constraints to be considered during operator dependency resolution. This feature will ensure Operator Lifecycle Manage (OLM) to install operators that satisfy dependency requirements and runtime constraints if present.
+
+## Motivation
+
+At the moment, OLM resolves a set of operators which seem that they will work together based on dependency requirements that are specifed by operator bundles. However, it is possible that those resolved operators will not work on the clusters due to cluster/runtime constraints. For example, if the cluster is running a specific Kubernetes version that is not compatible with the operators, the installed operators may fail to work propertly on that given cluster.
+
+The cluster runtime constraints are often unknown by the operator authors and should be specified by cluster admins. However, currently, there is no machanism for cluster admins to specify runtime constraints that OLM can understand and take under consideration during installation.
```

Comment:
I think it would help to use different terms for the two sides of this problem. For example you might say an operator has "requirements" and a cluster has "capabilities" or "properties". Similar to "taints" and "tolerations". Using the term "constraints" for both sides makes the discussion, to me at least, a bit harder to follow. I like the part where you identify the different personas (operator authors and cluster admins), so picking a different term to describe the information each persona provides will help clarify discussions of the workflow.